### PR TITLE
docs: adding env var info

### PR DESCRIPTION
**Pull request by trujillo-adam**
This PR adds information about environment variables required to use Consul Connect with Envoy when TLS is enabled.

I will make additional readability and formatting improvements to the docs in subsequent commits that are outside the scope of the issue.

I will create a separate PR to link the related tutorial to the docs changes in this PR.

### CODE REVIEW

File: website/content/docs/connect/proxies/envoy.mdx [hunk 1]
```diff
@@ -93,28 +87,40 @@ responsibility for correctly configuring Envoy and ensuring version support etc.
...
 
 -> **Note:** Prior to Consul 1.9.0 intentions were exclusively enforced
 per-connection (L4) using an `ext_authz` network filter.
 
+## Fetching Certificates
+
+Envoy will use the [`CONSUL_HTTP_TOKEN`](/commands#consul_http_token) and [`CONSUL_HTTP_ADDR`](/commands#consul_http_addr) environment variables to contact Consul to fetch certificates if the following conditions are met:
+
+- The `CONSUL_HTTP_TOKEN` environment variable contains a Consul ACL.
+- The Consul ACL has the necessary permissions to read configuration for that service.
```
Comment (blake) [inline website/content/docs/connect/proxies/envoy.mdx:103]:
```suggestion
- The `CONSUL_HTTP_TOKEN` environment variable contains a Consul ACL token.
- The Consul ACL token has the necessary permissions to read configuration for that service.
```

File: website/content/docs/connect/proxies/envoy.mdx [hunk 2]
```diff
@@ -93,28 +87,40 @@ responsibility for correctly configuring Envoy and ensuring version support etc.
...
+
+If TLS is enabled on Consul, you will also need to add the following environment variables _prior_ to starting Envoy:
+
+- [`CONSUL_CACERT`](/commands#consul_cacert)
+- [`CONSUL_CLIENT_CERT`](/commands#consul_client_cert)
+- [`CONSUL_CLIENT_KEY`](/commands#consul_client_key)
```
Comment (blake) [inline website/content/docs/connect/proxies/envoy.mdx:109]: According to the linked issue, the following parameters are also required.

```suggestion
- [`CONSUL_CLIENT_KEY`](/commands#consul_client_key)
- [`CONSUL_HTTP_SSL`](/commands#consul_http_ssl)
```

The issue also suggests that `CONSUL_HTTP_ADDR` and `CONSUL_GRPC_ADDR` be defined with the scheme set to `https://`, but according to the [`CONSUL_GRPC_ADDR`](https://www.consul.io/commands#consul_grpc_addr) docs it should be sufficient just to set `CONSUL_HTTP_SSL=true`.

**Review by blake — APPROVED** [review]
I had a few minor suggestions. Otherwise, LGTM.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/424601.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 067c8c2c5c4a2b006e2f40d69c2560f78a26c8d7 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 067c8c2c5c4a2b006e2f40d69c2560f78a26c8d7 onto `release/1.10.x` succeeded!

**Pull request merged**